pkg/controller/machineset/controller.go

```diff
@@ -68,7 +76,18 @@ func (c *MachineSetControllerImpl) Reconcile(machineSet *v1alpha1.MachineSet) er
 		return err
 	}
 
-	return c.syncReplicas(machineSet, filteredMachines)
+	syncErr := c.syncReplicas(machineSet, filteredMachines)
+
+	ms := machineSet.DeepCopy()
```

Comment:
Two questions:

1. Why is it OK for the code below to continue if syncErr contains an error?
2. Given #1 above, we can reach line 85 with syncErr populated with an error value. Given In the case of an error on line 85, will cause us to reach line 87 why is it OK that we could potentially lose an error value in "syncErr".

Reply:
The reason to continue is ensure the MachineSet Status that contains the number of replicas/labeled/ready/available is correct.
syncReplicas may have changed one of the counts in the status and it is worthwhile to update it.

You make a good point about possibly dropping the syncErr.
I changed it such that the two errors are combined.